### PR TITLE
Add MetadataControlImprovements testcases

### PR DESCRIPTION
**Pull request by maryvictol**
This PR  adds testcases for Metadata control improvements (#2023 ).

### CODE REVIEW

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_1-Column_sorting_extension.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as admin user from the prerequisites | |
+| 2 | Open the **Library** page | |
+| 3 | Open `Project1` created at step 3 of the preparations | |
+| 4 | Click the **Metadata** item | |
+| 5 | Click the *Sample [...]* item in the list | Sample metadata table opens |
+| 6 | Click on ***R1_Fastq*** column header | <li> Pictogram of decreasing appears near the ***R1_Fastq*** header <li> Table is sorted by ***R1_Fastq*** column in decreasing order |
+| 7 | Click on ***Sample Name*** column header | <li> Order number `1` appears near the decrising pictogam for ***R1_Fastq*** header <li> Pictogram of decreasing appears near the ***Sample Name*** header with order number `2` <li> Table is sorted by ***R1_Fastq*** column in decreasing order and by ***Sample Name*** column in decreasing order after that |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_1-Column_sorting_extension.md]: > the decrising pictogam for ***R1_Fastq*** header

fix to 'the decreasing pictogram for ***R1_Fastq*** header'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_2-Metadata_entities_display_setting.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+| 2 | Open the **Library** page | |
+| 3 | Open `Project1` created in [_2023_1_](2023_1-Column_sorting_extension.md) | |
+| 4 | Click the **Metadata** item | |
+| 5 | Click the *Sample [...]* item in the list | <li> Sample metadata table opens <li> ***Show columns*** button is shown near the Search field |
+| 6 | Click ***Show columns*** button | ***Show columns*** control is opened and contains a list of selectable metadata columns |
+| 7 | Uncheck checkboxes for ***Created Date*** and ***R2_Fastq*** columns | Unchecked columns aren't shown in the table |
+| 8 | Navigate to 2nd page of table | Unchecked columns aren't shown in the table |
+| 9 | Click ***Show columns*** button | |
+| 10 | Try to uncheck all checkboxes | Last unchecked checkbox is disabled |
+| 11 |  Move ***Sample Name*** row to the top of list | ***Sample Name*** is shown as first column in the table |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_2-Metadata_entities_display_setting.md]: 1. Add a step before this one - where all checkboxes should be ticked again.
2. Add a description "how to move" - e.g., 'Holding the icon near the checkbox, move the ***Sample Name*** row...'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Metadata entities autofill
+Test verifies
+- that the users can pre-configuring the display order and visibility of metadata columns for a large number of columns.
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Fix the case description

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+| Steps | Actions | Expected results |
+| :---: | --- | --- |
+| 1 | Login as admin user from the prerequisites | |
+| 2 | Open the **Library** page | |
+| 3 | Open `Project1` created in [_2023_1_](2023_1-Column_sorting_extension.md) | |
+| 4 | Click the **Metadata** item | |
+| 5 | Click the *Sample [...]* item in the list | <li> Sample metadata table opens <li> ***Show columns*** button is shown near the Search field |
+| 6 | Click any Sample | Attributes section opens |
+| 7 | Add a new key:value pair `BCL:BCL` | |
+| 8 | Click on a cell with value `BCL` and drag the fill handle on the several cells in the column | Selected cells are filled with value `BCL` |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: > Click on a cell with

Fix to 'Hover over a cell with'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+| 9 | Click on selected cells content menu | Content menu has items *Copy cells* and *Revert* |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: > Click on selected cells content menu

Add '(icon in the bottom corner of the selected area)'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+| 10 | Select *Revert* option | All values exclude first value are removed |
+| 11 | Change value for key:value pair `BCL:BCL` to `BCL004` | |
+| 12 | Repeat step 8 | Selected cells are filled with the following values in order |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Add to clarify - '... values in order - i.e. `BCL004`, `BCL005`, `BCL006`, etc.'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+| 13 | Click on selected cells content menu | Content menu has items *Fill cells*, *Copy cells* and *Revert* |
+| 14 | Select *Copy cells* option | Selected cells are filled with value `BCL004` |
+| 15 | Select *Fill cells* option | Selected cells are filled with the following values in order |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Add to clarify - '... values in order - i.e. `BCL004`, `BCL005`, `BCL006`, etc.'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+**Preparations**:
+
+1. Login as admin user from the prerequisites
+2. Open the **Library** page
+3. Create a new project (`Project1`), open it
+4. Hover over **+Create** button
+5. Select *Configuration* option
+6. Enter `config1` into the *Name* field and click **Create** button
+7. Fill required fields on the Configuration page
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: > Fill required fields

Add 'Open the `config1` configuration and fill ...'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+**Prerequisites**:
+
+- Admin user
+- Perform [_2023_1_](2023_1-Column_sorting_extension.md) case
+
+**Preparations**:
+
+1. Login as admin user from the prerequisites
+2. Open the **Library** page
+3. Create a new project (`Project1`), open it
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: > Create a new project (`Project1`)

Fix to 'Find a project created at [_2023_1_](2023_1-Column_sorting_extension.md) case (`Project1`) ...'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+| 13 | Click the **Rerun** hyperlink | *Launch* form opens |
+| 14 | Hover over arrow near the **Launch** button | Menu with option *Select metadata entries and launch* appears |
+| 15 | Select *Select metadata entries and launch* option | ***Select metadata*** modal appears |
+| 16 | Click the **Metadata** item. Click the *Sample* item in the list | Sample metadata table opens |
+| 17 | Select Sample not equal to `Sample1` (`Sample2`). Click **OK** button. Confirm launch |
+| 18 | Repeat steps 9-10 | Parameter `SampleName:` with value equals `Sample2` SampleName and Parameter `R1_Fastq:` with value equals `Sample2` R1_Fastq are shown in the ***Parameters*** section |
+| 19 | Go to the **Runs** page. Switch to the ***Completed Runs tab*** | |
+| 20 | Click the **Rerun** hyperlink for run with `run_id` from step 9 | *Launch* form opens |
+| 21 | Click **Launch** button. Confirm launch | |
+| 22 | Repeat steps 9-10 | Parameter `SampleName:` with value equals `Sample1` SampleName and Parameter `R1_Fastq:` are shown in the ***Parameters*** section |
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: > Parameter `R1_Fastq:` are shown

Fix to 'Parameter `R1_Fastq:` with value equals `Sample1` R1_Fastq are shown'

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+| 23 | Repeat steps 11-13 | *Launch* form opens <li> arrow isn't shown near the **Launch** button |
+
+***NOTE:*** Stops all runs.
```
Comment (NShaforostov) [inline docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: Fix to 'Stop all runs'

**Review by NShaforostov — COMMENTED** [review]
Fix minor remarks. See comments, please.

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_1-Column_sorting_extension.md (hunk 1 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_1-Column_sorting_extension.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md (hunk 11 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md (hunk 10 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md (hunk 9 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md (hunk 8 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_5-Samples_preselect_for_a_rerun.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md (hunk 7 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md (hunk 6 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md (hunk 4 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md (hunk 5 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_2-Metadata_entities_display_setting.md (hunk 2 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_2-Metadata_entities_display_setting.md]: Done

File: docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md (hunk 3 above)
Comment (maryvictol) [reply docs/testcases/UI/Sample_Processing_Module/2023_Metadata_control_improvements/2023_3-Metadata_entities_autofill.md]: Done

**Review by NShaforostov — APPROVED** [review]

**Pull request merged**